### ISSUE TITLE
Add Tickets (offers)

### ISSUE THREAD

**Issue by saumier** [culturecreates/footlight-app#29]
Add Tickets

There are 2 states for the tickets: free, paid.

When you first create an event, you choose either free or paid.  A choice is required in order to publish the event.

![Image](https://user-images.githubusercontent.com/419491/211912125-fbd54943-d121-4f86-b57a-522e3bad8caf.png)

If you choose free, then the button is selected, and the notes field (en/fr) appears. The note is optional.  You can switch to Paid by clicking the paid button. When you exit and return to edit the event, the UI should reflect the chosen state, meaning if you chose free then when you return to edit the event the free button is already selected.

![Image](https://user-images.githubusercontent.com/419491/211912161-37abb28f-e0bd-4ff8-a801-b1e37dfc4546.png)
warning - in the above image, the note field should have en/fr

When you select Paid, there is a Buy ticket link field. Next is a single row (only show one row by default) to enter the optional price and optional price description (en/fr). Below that there is the optional note (en/fr) field. 

You can click "+ another ticket type" to add additional ticket type rows so you can enter more prices and price descriptions. You can delete rows of ticket prices.  The prices must be a number with up to 2 decimals (13 or 13.50) and no letters.

For paid tickets, there must be either a buy ticket url or a price. So a paid ticket is valid if it has only a buy ticket url or only a price.

![Image](https://user-images.githubusercontent.com/419491/211911718-117ad22c-1cee-4624-9ae0-8dd35d5e0b93.png)

**Comment by saumier** [comment]
@SuhailAliyar The validation for tickets should be changed.  A ticket is required to be
free or paid. A paid ticket only needs a buy ticket url. Price is not mandatory and is often missing.

Current code:
 (!offers?.length
        && !(offers?.some(offer => offer.type == OfferType.OFFER || (OfferType.AGGREGATE_OFFER && offer.price))))


Expected code something like:
// Checks free event
const free = (offer) => offer.type === OfferType.AGGREGATE_OFFER && offer.price === 0  
// Checks paid event
const paid = (offer) => offer.type === OfferType.AGGREGATE_OFFER && offer.url

!offers.some(offer => free || paid)

**Comment by saumier** [comment]
@troughc Please test is staging.

**Comment by troughc** [comment]
Test in Chrome on this event: http://staging.cms.footlight.io/dashboard/62df90a4820e41f7c1359760/events/add-event/63b72a43d25d4c007098562c
(La Poesie....)
It's looking really good, everything seems to be working - I reviewed the published results on demo staging here: http://demo.staging-tout-culture.s3-website.ca-central-1.amazonaws.com/events/event-details.html?eventId=63b72a43d25d4c007098562c 

@saumier One thing: the note contents aren't cleared out when a user switches from Paid to Free or Free to Paid. So a user would have to remember to update what is there.  Let's discuss before deciding on what is the right action.

**Comment by troughc** [comment]
@saumier please take a look.

**Comment by troughc** [comment]
First problematic case retested: PASS
on comment, it would be good to add an automatic scroll to the first instance of a problem (in this case, scroll to the ticket field). 
Second problematic case retested
I am still encountering a bug
(example event: Ticket test 3): I followed the steps to create the issue. Step 3 (save) works now. So after step 3:
4. Go back into event and click Publish. Get a warning (good).
5. Add a link to the field. Click Publish - nothing happens (not good). the publish button is grayed out, but no message below
6. Workaround: click save. this works and you are back in event list. Then reenter the event (with the ticket link) and click Publish. It now works.

**Comment by saumier** [comment]
@troughc I am not sure if its a good UX to scroll to the first instance of a problem.  Have you considered adding a popup message whenever the user tries to save an event and there is an error in the form? The message could list the problem so the user knows which parts of the form need attention.

**Comment by troughc** [comment]
Tested new version: PASSED.
